### DISCUSSION TITLE
New TSML UI feature: Speaker/Discussion

### DISCUSSION THREAD

**Discussion by joshreisner** [code4recovery/12-step-meeting-list#1004] [Show and tell]
Announcing a minor update to TSML UI: 

> Now, when meetings are tagged both Speaker (`SP`) and Discussion (`D`), TSML UI merges them into a combined Speaker/Discussion type. This enables users to use existing filters to locate Speaker-only and Discussion-only meetings.

For an example, see: https://aasfmarin.org/meetings

Previously, selecting "Speaker" or "Discussion" types would include meetings with one or both types applied, making them less useful to people looking for a specific type of meeting.

**Comment by Grateful-Viv** [comment]
If I understand this, you’re adding another type “speaker/discussion” and keeping discussion only and speaker only. Correct?

**Reply by joshreisner** [reply]
yes but it's more that we're changing the way we render meetings that have both the Speaker and Discussion type. if a meeting has both, then it looks like they have a merged Speaker/Discussion type

no changes if the meeting is Speaker-only or Discussion-only

**Reply by Grateful-Viv** [reply]
So there’s no new type to select from, but if I select speaker and also select discussion, it will be displayed as speaker/discussion? 

Will it show up in the meeting codes as SP and also D?

**Reply by joshreisner** [reply]
yes and yes

the meeting guide app is not affected. i will be recommending they make this update as well, but we shall see if they do
